### DISCUSSION TITLE
Use ix.Platform() to open a local database

### DISCUSSION THREAD

**Discussion by Chocooto** [iiasa/message_ix#440] [Q&A] [answered]
Message-ix and ixmp are updated to the latest version 3.2.0.
I can use ix.Platform() or ix.Platform('default') to open a default database from a default address. How can I open another database from a different address? Thanks!
I tried different ways but failed, and errors are as following.
**1. mp = ix.Platform(dbprops="E:\Scenarios\GEIDCO-2C\geidco2")**

> FileNotFoundError                         Traceback (most recent call last)
> <ipython-input-20-c61f5a3432eb> in <module>
> ----> 1 mp = ix.Platform(dbprops="E:\Scenarios\GEIDCO-2C\geidco2")
> 
> ~\anaconda3\lib\site-packages\ixmp\core.py in __init__(self, name, backend, **backend_args)
>      97 
>      98         # Instantiate the backend
> ---> 99         self._backend = backend_class(**kwargs)
>     100 
>     101     def __getattr__(self, name):
> 
> ~\anaconda3\lib\site-packages\ixmp\backend\jdbc.py in __init__(self, jvmargs, **kwargs)
>     223                     raise ValueError("Config file contains no database URL")
>     224             else:
> --> 225                 raise FileNotFoundError(dbprops)
>     226 
>     227         start_jvm(jvmargs)
> 
> FileNotFoundError: E:\Scenarios\GEIDCO-2C\geidco2

**2.mp = ix.Platform('geidco2')**

> KeyError                                  Traceback (most recent call last)
> ~\anaconda3\lib\site-packages\ixmp\_config.py in get_platform_info(self, name)
>     303         try:
> --> 304             return name, copy(self.values["platform"][name])
>     305         except KeyError:
> 
> KeyError: 'geidco2'
> 
> During handling of the above exception, another exception occurred:
> 
> ValueError                                Traceback (most recent call last)
> <ipython-input-21-a90d3a13bc92> in <module>
> ----> 1 mp = ix.Platform('geidco2')
> 
> ~\anaconda3\lib\site-packages\ixmp\core.py in __init__(self, name, backend, **backend_args)
>      81         if name:
>      82             # Using a named platform config; retrieve it
> ---> 83             self.name, kwargs = config.get_platform_info(name)
>      84 
>      85         # Overwrite any platform config with explicit keyword arguments
> 
> ~\anaconda3\lib\site-packages\ixmp\_config.py in get_platform_info(self, name)
>     304             return name, copy(self.values["platform"][name])
>     305         except KeyError:
> --> 306             raise ValueError(
>     307                 f"platform name {repr(name)} not among "
>     308                 + repr(sorted(self.values["platform"].keys()))
> 
> ValueError: platform name 'geidco2' not among ['default', 'local']
> from None

**Comment by khaeru** [comment, marked as answer]
The second method is the current one. However:

> ValueError: platform name 'geidco2' not among ['default', 'local']

…it looks like you haven't stored the location of this 'geidco2' local database in your ixmp configuration file. Please see
https://docs.messageix.org/projects/ixmp/en/stable/api.html#configuration, in particular the command `ixmp platform add`.

**Reply by Chocooto** [reply]
Thanks a lot!

**Reply by khaeru** [reply]
Please click the ✔️  icon on any answer(s) that helped with your question!

**Comment by francescolovat** [comment]
To complement the comment above, in order for you to create a new platform and make it work using `mp = ix.Platform('geidco2')`, you should run in your terminal (e.g. Anaconda prompt) the following:
`ixmp platform add geidco2 jdbc hsqldb E:\Scenarios\GEIDCO-2C\geidco2`

**Reply by Chocooto** [reply]
Thanks a lot!
